code/modules/food_and_drinks/food/pizza.dm

```diff
@@ -0,0 +1,361 @@
+
+/////////////////////////////////////////////////PIZZA////////////////////////////////////////
+
+/obj/item/food/pizza
+	icon = 'icons/obj/food/pizzaspaghetti.dmi'
+	w_class = WEIGHT_CLASS_NORMAL
+	max_volume = 80
+	food_reagents = list(/datum/reagent/consumable/nutriment = 28, /datum/reagent/consumable/tomatojuice = 6, /datum/reagent/consumable/nutriment/vitamin = 5)
+	tastes = list("crust" = 1, "tomato" = 1, "cheese" = 1)
+	foodtypes = GRAIN | DAIRY | VEGETABLES
+	burns_in_oven = TRUE
+	/// type is spawned 6 at a time and replaces this pizza when processed by cutting tool
+	var/obj/item/food/pizzaslice/slice_type
+	///What label pizza boxes use if this pizza spawns in them.
+	var/boxtag = ""
+
+/obj/item/food/pizza/MakeProcessable()
+	if (slice_type)
+		AddElement(/datum/element/processable, TOOL_KNIFE, slice_type, 6, 30)
+
+/obj/item/food/pizza/raw
+	foodtypes =  GRAIN | DAIRY | VEGETABLES | RAW
+	burns_in_oven = FALSE
+	slice_type = null
+
+/obj/item/food/pizza/raw/MakeBakeable()
+	AddComponent(/datum/component/bakeable, /obj/item/food/pizza, rand(70 SECONDS, 80 SECONDS), TRUE, TRUE)
+
+/obj/item/food/pizzaslice
+	icon = 'icons/obj/food/pizzaspaghetti.dmi'
+	food_reagents = list(/datum/reagent/consumable/nutriment = 5)
+	foodtypes = GRAIN | DAIRY | VEGETABLES
+	w_class = WEIGHT_CLASS_SMALL
+
+
+/obj/item/food/pizzaslice/Initialize()
+	. = ..()
+	//AddElement(/datum/element/processable, TOOL_ROLLINGPIN, /obj/item/stack/sheet/pizza, 1, 10)
+
+/obj/item/food/pizza/margherita
+	name = "pizza margherita"
+	desc = "The most cheezy pizza in galaxy."
+	icon_state = "pizzamargherita"
+	food_reagents = list(/datum/reagent/consumable/nutriment = 25, /datum/reagent/consumable/tomatojuice = 6, /datum/reagent/consumable/nutriment/vitamin = 5)
+	tastes = list("crust" = 1, "tomato" = 1, "cheese" = 1)
+	tastes = list("crust" = 1, "tomato" = 1, "cheese" = 1)
+	foodtypes = GRAIN | VEGETABLES | DAIRY
+	slice_type = /obj/item/food/pizzaslice/margherita
+
+/obj/item/food/pizza/margherita/Initialize()
```

Comment:
Redundant due to the base typepath having a generic method, would actually create double.